core/src/main/java/org/apache/druid/common/utils/SocketUtil.java

```diff
@@ -41,24 +41,11 @@ public static int findOpenPortFrom(int startPort)
     int currPort = startPort;
 
     while (currPort < 0xffff) {
-      ServerSocket socket = null;
-      try {
-        socket = new ServerSocket(currPort);
+      try (ServerSocket socket = new ServerSocket(currPort)) {
```

Comment:
The behavior is a bit different. IOExceptions on `socket.close()` used to be suppressed, but now they'll be thrown. I think that's okay, since we don't expect to get exceptions when closing the socket, but just wanted to point it out.